### PR TITLE
Dryrun cleanup

### PR DESCRIPTION
**Pull request by aojea**
The feature is GA since 1.19

https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

/kind cleanup
```release-note
NONE
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1125,7 +1125,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	genericfeatures.CustomResourceValidationExpressions: {Default: true, PreRelease: featuregate.Beta},
 
-	genericfeatures.DryRun: {Default: true, PreRelease: featuregate.GA},
+	genericfeatures.DryRun: {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```
Comment (liggitt) [inline pkg/features/kube_features.go:1128]: comment to remove in 1.28

File: staging/src/k8s.io/apiserver/pkg/features/kube_features.go [hunk 2]
```diff
@@ -208,7 +208,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	CustomResourceValidationExpressions: {Default: true, PreRelease: featuregate.Beta},
 
-	DryRun: {Default: true, PreRelease: featuregate.GA},
+	DryRun: {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go:211]: comment to remove in 1.28

File: pkg/features/kube_features.go (hunk 1 above)
Comment (aojea) [reply pkg/features/kube_features.go:1128]: done

File: staging/src/k8s.io/apiserver/pkg/features/kube_features.go (hunk 2 above)
Comment (aojea) [reply staging/src/k8s.io/apiserver/pkg/features/kube_features.go:211]: done

**Comment by aojea** [comment]
hmm, how are we doing on terms of flakes?

https://storage.googleapis.com/k8s-triage/index.html?ci=0&pr=1

I start to see random and unrelated test failures

**Comment by aojea** [comment]
the apiserver fails to start here `pull-kubernetes-e2e-gce-alpha-features` that may be easily related, there are no apiserver logs though .... before start the digging @liggitt do you have any idea from the top of your head where to look at?

**Comment by liggitt** [comment]
Check the master kubelet log… often that means an alpha feature broke the kubelet that runs the apiserver

**Comment by apelisse** [comment]
Thanks for working on this!

**Comment by liggitt** [comment]
looks like the last green alpha e2e run was 8/23 (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/111616/pull-kubernetes-e2e-gce-alpha-features/1562230740056281088) so something merged in the last ~3 days that broke that job

**Comment by aojea** [comment]
Missing lgtm

**Comment by aojea** [comment]
/retest


Kubernetes e2e suite: [It] [sig-node] Pods should run through the lifecycle of Pods and PodStatus [Conformance] expand_more	1m9s

**Pull request merged**